src/main/java/RecrBad.java

```diff
@@ -39,28 +52,47 @@ private static void sayBye() {
         int randNum = rand.nextInt(upperbound);
 
         System.out.println("Adios My Friend. Sleep early, study smarter \n" + quotes[randNum]);
+        printLine();
     }
 
-    public static void main(String[] args) {
-        Scanner in = new Scanner(System.in);
-        String logo = " ____   ___    ___   ____    ___   \n"
-                + "| __/  / _ \\  | _ \\  | __|  / _ \\   \n"
-                + "| |    ||_||  | / /  | |_   | | |      \n"
-                + "| |_   | _ |  | _/   |_  /  | | |      \n"
-                + "|___\\  // \\\\  |_|    /__/ . \\___/  \n";
+    private static String[] addToList(String[] lines, String line) {
+        // copies and returns longer String[lines.length+1]
+        String[] moreLines = Arrays.copyOf(lines, lines.length + 1);
+        moreLines[lines.length] = line; //append at last elem
 
-        System.out.println("Hello! I'm \n" + logo);
         printLine();
-        System.out.println("What can I do for you?");
+        System.out.println("You added: " + line);
+        return moreLines;
+    }
+
+    private static void displayList(String[] lines) {
+        int count = 1;
+        for (String line : lines) {
+            System.out.println(count + ". " + line);
+            count += 1;
+        }
+    }
+
+    public static void main(String[] args) {
+        Scanner in = new Scanner(System.in);
+        sayHi();
+        String[] lines = new String[]{};
 
         while (true) {
             printLine();
             String line = in.nextLine(); // reads input
+
             if (line.toUpperCase().equals("BYE")) {
-                sayBye();
+                sayBye(); // Ctrl B to see def, shift F10 to run, Ctrl Alt L reformat
```

Comment:
Comments were at appropriate places which made it easier to read and understand the code 👍